daemon/logs.go

```diff
@@ -58,7 +55,21 @@ func (daemon *Daemon) ContainerLogs(container *Container, config *ContainerLogsC
 		tailLines = -1
 	}
 
+	outStream := config.OutStream
+	errStream := outStream
+
+	// write an empty chunk of data (this is to ensure that the
+	// HTTP Response is sent immediately, even if the container has
+	// not yet produced any data)
+	outStream.Write(nil)
+
 	logrus.Debug("logs: begin stream")
+	if !container.Config.Tty {
+		errStream = stdcopy.NewStdWriter(outStream, stdcopy.Stderr)
+		outStream = stdcopy.NewStdWriter(outStream, stdcopy.Stdout)
+	}
+	config.OutStream = outStream
```

Comment:
It's a dead write if I take out the fprintf in api/server/container.go

Yea, since I'm looking at the attach code, it looks like this block was in the past copied and pasted from there, where thy do write it back before reading it out again. I'll change it here.